lib/matchRoutes.js

```diff
@@ -103,12 +104,23 @@ function matchRoutes(routes, path, routerURLPatternOptions) {
  *
  * @private
  */
-function Match(path, route, match, query) {
+function Match(path, route, match, query, valueDecoders) {
   this.path = path;
   this.route = route;
   this.match = match;
   this.query = query;
 
+  if (valueDecoders) {    
+    Object.keys(match || {}).forEach(function(key) {
+      var fn = valueDecoders[key];
```

Comment:
Sure, but there could be a key on `match` that's a builtin on `Function.prototype`.